profiles/seccomp/seccomp.go

```diff
@@ -6,28 +6,64 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/versions/v1p24"
 	"github.com/opencontainers/specs/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
 )
 
 //go:generate go run -tags 'seccomp' generate.go
 
 // GetDefaultProfile returns the default seccomp profile.
 func GetDefaultProfile(rs *specs.Spec) (*specs.Seccomp, error) {
-	return setupSeccomp(DefaultProfile(rs))
+	return setupSeccomp(DefaultProfile(), rs)
 }
 
 // LoadProfile takes a file path and decodes the seccomp profile.
-func LoadProfile(body string) (*specs.Seccomp, error) {
-	var config types.Seccomp
+func LoadProfile(body string, rs *specs.Spec) (*specs.Seccomp, error) {
+	var (
+		config types.Seccomp
+		err    error
+	)
 	if err := json.Unmarshal([]byte(body), &config); err != nil {
 		return nil, fmt.Errorf("Decoding seccomp profile failed: %v", err)
 	}
+	// try with the old profile
+	// TODO(runcom): this is not going to work properly:
+	// if I use Version == 0 in a new format seccomp profile, what's going to
+	// happen?
+	if config.Version == 0 {
```

Comment:
right, but the issue it's still there (regardless of both `names` and `name` supported - also, ppl can write both which can be ok but cannot as well). 
@justincormack can you tell how do you believe it should behave in the following cases (expecially 1) and 2)):

-

1) No version (which means 0 and fallback to `decodeLegacySeccomp`), new format with `names` and the v0 struct json doesn't support `names` so syscalls in this profile will be silently ignored 
```
{
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"names": ["chmod"],
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
2) same as above:
```
{
	"verison": 0,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"names": ["chmod"],
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
3) works as expected - fallback to v0 decoder + syscall applied
```
{
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"name": "chmod",
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
4) same as above
```
{
	"version": 0,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"name": "chmod",
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
5) works as expected - v1 seccomp profile + `names`
```
{
	"version": 1,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"names": ["chmod"],
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
6) works as expected - v1 profile + `name` (singular)
```
{
	"version": 1,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"name": "chmod",
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```
7) works somehow as expected, blocks both `unshare` and `chmod`
```
{
	"version": 1,
	"defaultAction": "SCMP_ACT_ALLOW",
	"syscalls": [
		{
			"name": "chmod",
			"names": ["unshare"],
			"action": "SCMP_ACT_ERRNO"
		}
	]
}
```